### PR TITLE
[GAUDISW-244575] Adapt OnlineDefragmenter and CacheSwapUtils for torc…

### PR DESCRIPTION
**Pull request by jwieczorekhabana**
…h.compile

Because of the double entrypoint of CacheSwapUtils (forward and swap functions) torch.compile would process module and forward function while swap's self would refer to unwrapped module. That results in the function not being run as compiled Changes made in this patch:
 - Hide CacheSwapUtils entirely in OnlineDefragmenter. Let it be responsible for calling the module correctly
 - Moved warmup_defragmenter to defragmenter itself
 - Removed initialize function of OnlineDefragmenter, fully initialize object in __init__
 - Adapted unit tests for new implementations

### CODE REVIEW

File: vllm_gaudi/extension/defragmentation.py [hunk 1]
```diff
@@ -201,12 +184,64 @@ def defragment(self):
...
+        dsts = pad_list(list(dsts), threshold, itertools.repeat(-1))
+        srcs = torch.tensor(srcs, dtype=torch.long, device='cpu').to('hpu', non_blocking=True)
+        dsts = torch.tensor(dsts, dtype=torch.long, device='cpu').to('hpu', non_blocking=True)
+        key_caches = [cache[0] for cache in self.kv_caches]
+        self.cache_utils(srcs, dsts, key_caches, self.block_size)
+        if not self.is_mla:
+            value_caches = [cache[1] for cache in self.kv_caches]
+            self.cache_utils(srcs, dsts, value_caches, self.block_size)
+
+    def warmup(self):
```
[the remark this answers is not in the document]
Comment (jwieczorekhabana) [reply vllm_gaudi/extension/defragmentation.py]: It's warmed up in exactly the same place as before. Just moved the function implementing warmup to defragmenter
